src/utils/supabase.ts

```diff
@@ -841,3 +845,98 @@ export const getChannel = async (
     message: "Successfully fetched the channel!",
   };
 };
+
+export const getTutors = async (
+  query: TutorQuery
+): Promise<SupabaseResponse<TutorSelect[]>> => {
+  const tutorsResponse = await supabase
+  .from("tutors")
+  .select("*")
+  .eq("contact_id", query.contact_id);
+
+  if (tutorsResponse.error) {
+    return {
+      error: true,
+      message: "There was an error fetching tutors!"
+    };
+  }
+
+  if (tutorsResponse.data.length == 0) {
+    return {
+      error: true,
+      message: "No tutors were found!"
+    };
+  }
+
+  console.log(tutorsResponse);
+
+  return {
+    data: tutorsResponse.data,
+    error: false,
+    message: "Successfully fetched tutors!"
+  };
+};
+
+export const getTutor = async (
+  queryData: UniqueTutorQuery
+): Promise<SupabaseResponse<TutorSelect>> => {
+  const tutorResponse = await getTutors(queryData);
+
+  if (tutorResponse.error) {
+    return tutorResponse;
+  }
+
+  return {
+    ...tutorResponse,
+    data: tutorResponse.data[0],
+  };
+};
+
+export const getTutorId = async (
+  queryData: UniqueTutorQuery
+): Promise<SupabaseResponse<string>> => {
+  if (queryData.tutor_id) {
+    return {
+      data: queryData.tutor_id,
+      error: false,
+      message: "Tutor ID already exists!",
+    };
+  }
+
+  const tutorResponse = await getTutor(queryData);
+
+  if (tutorResponse.error) {
+    return tutorResponse;
+  }
+
+  const { tutor_id } = tutorResponse.data;
+
+  return {
+    data: tutor_id,
+    error: false,
+    message: "Successfully fetched Tutor ID!",
+  };
+};
+
+export const insertTutorLog = async (
+  tutorLogInfo: TutorLogInsert
+): Promise<SupabaseResponse<TutorLogSelect>> => {
+  const tutorLogResponse = await supabase
+    .from("tutor_logs")
+    .insert(tutorLogInfo)
+    .select();
+
+  if (tutorLogResponse.error) {
+    return {
+      error: true,
+      message: tutorLogResponse.error,
```

Comment:
I'd suggest creating a custom error message so that the end user isn't getting Supabase error messages. Something like "There was an error inserting the tutor log!" should be fine.